content/blog/2017/05/2017-05-17-pipeline-dev-tools.adoc

```diff
@@ -0,0 +1,85 @@
+---
+layout: post
+title: "Pipeline Development Tools"
+tags:
+- blueocean
+- pipeline
+- tutorial
+
+author: lnewman
+---
+
+NOTE: This is a guest post by link:https://github.com/bitwiseman[Liam Newman],
+Technical Evangelist at link:https://cloudbees.com[CloudBees].
+
+== Pipeline as code, made easier
+
+I've only been working with Pipeline for about a year.
+Pipeline in and of itself is a huge improvement over old style Jenkins projects.
+As a developer, it has been so great be able work with Jenkins projects
+using the same tools I use for writing any other kind of code.
+
+I've also found a number of tools that are super helpful specifically
+for developing pipelines. Some were easy to find like the
+<</doc/book/pipeline/getting-started#built-in-documentation, built-in documentation>>
+and the
+<</doc/book/pipeline/getting-started#snippet-generator, Snippet Generator>>.
+Others were not as obvious or were only recently released.
+In this post, I'll show how a few of those tools make working with Pipelines
+even better.
+
+
+== The Blue Ocean Pipeline Editor
+
+The best way to start this list is with the most recent and coolest
+arrival in this space: the Blue Ocean Pipeline Editor.  The editor only works
+with Declarative Pipelines, but it brings a sleek new user experience to writing
+Pipelines.  My recent screencast, released as part of the Blue Ocean Launch,
+gives good sense of how useful the editor is:
+
+video::5Nct-jrZBbM[youtube]
+
+== Command-line Pipeline Linter
+
+One of the neat features of the Blue Ocean Pipeline Editor is that it does basic
+validation on our Declarative Pipelines before they are even committed or Run.
+This feature is based on the
+"link:https://en.wikipedia.org/wiki/Lint_(software)[Declarative Pipeline Linter]",
```

Comment:
I would expect this link to go to documentation, not a wiki page about generalized linting